File: testdata/bench/kotlin/BenchmarkTest00008.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via receiveText body
suspend fun handler00008(call: ApplicationCall) {
    val filter = call.receiveText()
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    conn.createStatement().executeQuery("SELECT * FROM logs WHERE message LIKE '%${filter}%'")
    call.respondText("OK")
}

File: testdata/bench/go/BenchmarkTest00008.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via multiple assignments
func Handler00008(w http.ResponseWriter, r *http.Request) {
	raw := r.URL.Query().Get("filter")
	filter := raw
	clause := "WHERE status = '" + filter + "'"
	query := "SELECT * FROM orders " + clause
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query(query)
}
